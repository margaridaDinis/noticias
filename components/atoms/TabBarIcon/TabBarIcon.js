import * as React from 'react';
import { Ionicons } from '@expo/vector-icons';
import { GUTTER } from '../../../constants/Layout';

export default function TabBarIcon(props) {
  return (
    <Ionicons
      {...props}
      size={GUTTER * 2}
      style={{ marginBottom: -(GUTTER / 4) }}
      color={props.color}
    />
  );
}
