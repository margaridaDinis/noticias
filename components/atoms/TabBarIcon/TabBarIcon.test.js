import React from 'react';
import { render } from '@testing-library/react-native';
import TabBarIcon from './TabBarIcon';

describe('<TabBarIcon />', () => {
  it('renders without props', () => {
    const { baseElement } = render(<TabBarIcon />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with custom props', () => {
    const { baseElement } = render(<TabBarIcon color="gold" />);

    expect(baseElement).toMatchSnapshot();
  });
});
