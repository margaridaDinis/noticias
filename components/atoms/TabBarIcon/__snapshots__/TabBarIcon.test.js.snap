// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBarIcon /> renders with custom props 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;

exports[`<TabBarIcon /> renders without props 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
